frontend/pages/policies/ManagePoliciesPage/components/RunScriptModal/RunScriptModal.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useCallback, useState } from "react";
+import { useQuery } from "react-query";
+import { omit } from "lodash";
+
+import { DEFAULT_USE_QUERY_OPTIONS } from "utilities/constants";
+
+import scriptsAPI, {
+  IListScriptsQueryKey,
+  IScriptsResponse,
+} from "services/entities/scripts";
+
+import { IPolicyStats } from "interfaces/policy";
+import { IScript } from "interfaces/script";
+
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import Checkbox from "components/forms/fields/Checkbox";
+import TooltipTruncatedText from "components/TooltipTruncatedText";
+import DataError from "components/DataError";
+import Spinner from "components/Spinner";
+import CustomLink from "components/CustomLink";
+import Button from "components/buttons/Button";
+import Modal from "components/Modal";
+
+// name avoids class name conflicts with hosts/details/HostDetailsPage/modals/RunScriptModal
+const baseClass = "policy-run-script-modal";
+
+interface IScriptDropdownField {
+  name: string;
+  value: number;
+}
+
+interface IFormPolicy {
+  name: string;
+  id: number;
+  runScriptEnabled: boolean;
+  scriptIdToRun?: number;
+}
+
+export type IPolicyRunScriptFormData = IFormPolicy[];
+
+interface IRunScriptModal {
+  onExit: () => void;
+  onSubmit: (formData: IPolicyRunScriptFormData) => void;
+  isUpdating: boolean;
+  policies: IPolicyStats[];
+  teamId: number;
+}
+
+const RunScriptModal = ({
+  onExit,
+  onSubmit,
+  isUpdating,
+  policies,
+  teamId,
+}: IRunScriptModal) => {
+  const [formData, setFormData] = useState<IPolicyRunScriptFormData>(
+    policies.map((policy) => ({
+      name: policy.name,
+      id: policy.id,
+      runScriptEnabled: !!policy.run_script,
+      scriptIdToRun: policy.run_script?.id,
+    }))
+  );
+
+  const anyEnabledWithoutSelection = formData.some(
+    (policy) => policy.runScriptEnabled && !policy.scriptIdToRun
+  );
+
+  const {
+    data: availableScripts,
+    isLoading: isLoadingAvailableScripts,
+    isError: isAvailableScriptsError,
+  } = useQuery<IScriptsResponse, Error, IScript[], [IListScriptsQueryKey]>(
+    [
+      {
+        scope: "scripts",
+        team_id: teamId,
+      },
+    ],
+    ({ queryKey: [queryKey] }) =>
+      scriptsAPI.getScripts(omit(queryKey, "scope")),
```

Comment:
Does this scope scripts to just the ones on the associated team?

Reply:
exactly

Reply:
the "scope" field of the above query key object is kind of just for readability, since the `getScripts` service method just takes `team_id` ( and/or `page` and/or `per_page`), so we omit it when passing in. Could probably just have the query key be `[{team_id: teamId}]`, but including `scope` reduces the likelihood of naming collision with other `useQuery` query keys on the same page, should that ever come up. (useQuery uses `query key` diffs to know when to automatically refetch...which yes we could use WAY more efficiently...similar to above `useMutation` discussion....)

Reply:
Got it. Main reason for the call-out was to check that we weren't cross-pollinating scripts from other teams as that should break on the API side, so given that we're doing that we're good.